src/atopile/cli/inspect.py

```diff
@@ -32,68 +33,59 @@
-                # Recursively visit the targets connected to this target
-                collect_connected_dfs(connections, connection.target.addr, visited)
-        elif connection.target.addr == root_addr:
-            if connection.source.addr not in visited:
-                visited.append(connection.source.addr)  # Mark this target as visited
-                # Recursively visit the targets connected to this target
-                collect_connected_dfs(connections, connection.source.addr, visited)
-    return visited
-
-def find_nets(graph) -> list[list[AddrStr]]:
     """
-    For a dict of connections:
-    {
-    A: [B, C],
-    B: [A, C],
-    C: [A, B],
-    D: [E],
-    E: [D]
-    }
-    This function returns the nets:
-    [[A, B, C], [D, E]]
+    This class represents the nets that are below the inspected module,
+    the equivalent net that is below the context module and
+    the individual connections that are made to the inspect net and the context net.
     """
-    visited = set()  # Keep track of visited nodes to avoid revisiting them.
-    connected_components = []  # Store the connected components.
+    def __init__(self, net: list[list[AddrStr]]):
+        self.inspect_net: list[AddrStr] = net
+        self.inspect_consumer: list[AddrStr] = []
+        self.context_net: list[AddrStr] = []
+        self.context_consumer: list[AddrStr] = []
+
+def find_nets_new(links: list[Link]) -> list[list[AddrStr]]:
+    # Convert links to an adjacency list
+    graph = {}
+    for link in links:
+        source = []
+        target = []
+        if match_interfaces(link.source.addr) and match_interfaces(link.target.addr):
+            for int_pin in get_children(link.source.addr):
+                if match_pins_and_signals(int_pin):
+                    source.append(int_pin)
+                    target.append(add_instance(link.target.addr, get_name(int_pin)))
+                else:
+                    raise errors.AtoNotImplementedError("Cannot nest interfaces yet.")
+        elif match_interfaces(link.source.addr) or match_interfaces(link.target.addr):
+            # If only one of the nodes is an interface, then we need to throw an error
+            raise errors.AtoTypeError.from_ctx(
+                link.src_ctx,
+                f"Cannot connect an interface to a non-interface: {link.source.addr} ~ {link.target.addr}"
+            )
+        # just a single link
+        else:
+            source.append(link.source.addr)
+            target.append(link.target.addr)
+
+        for source, target in zip(source, target):
+            if source not in graph:
+                graph[source] = []
+            if target not in graph:
+                graph[target] = []
+            graph[source].append(target)
+            graph[target].append(source)
+
     def dfs(node, component):
-        """
-        Perform depth-first search recursively to find all nodes connected to the current node.
-        """
         visited.add(node)
```

Comment:
I'm perhaps missing them but this seems to be using variables declared beneath itself?